tests/integration/security/jwt_test.go

```diff
@@ -464,83 +466,103 @@ func TestIngressRequestAuthentication(t *testing.T) {
-						},
-						{
-							Name:               "allow healthz",
-							Host:               "example.com",
-							Path:               "/healthz",
-							ExpectResponseCode: 200,
-						},
-					}
-
-					for _, c := range ingTestCases {
-						t.NewSubTest(c.Name).Run(func(t framework.TestContext) {
-							authn.CheckIngressOrFail(t, ingr, c.Host, c.Path, nil, c.Token, c.ExpectResponseCode)
-						})
-					}
 				})
 			}
+
+			// These test cases verify requests go through ingress will be checked for validate token.
+			ingTestCases := []struct {
+				Name               string
+				Host               string
+				Path               string
+				Token              string
+				ExpectResponseCode int
+			}{
+				{
+					Name:               "deny without token",
+					Host:               "example.com",
+					Path:               "/",
+					ExpectResponseCode: 403,
+				},
+				{
+					Name:               "allow with sub-1 token",
+					Host:               "example.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer1,
+					ExpectResponseCode: 200,
+				},
+				{
+					Name:               "deny with sub-2 token",
+					Host:               "example.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer2,
+					ExpectResponseCode: 403,
+				},
+				{
+					Name:               "deny with expired token",
+					Host:               "example.com",
+					Path:               "/",
+					Token:              jwt.TokenExpired,
+					ExpectResponseCode: 401,
+				},
+				{
+					Name:               "allow with sub-1 token on any.com",
+					Host:               "any-request-principlal-ok.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer1,
+					ExpectResponseCode: 200,
+				},
+				{
+					Name:               "allow with sub-2 token on any.com",
+					Host:               "any-request-principlal-ok.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer2,
+					ExpectResponseCode: 200,
+				},
+				{
+					Name:               "deny without token on any.com",
+					Host:               "any-request-principlal-ok.com",
+					Path:               "/",
+					ExpectResponseCode: 403,
+				},
+				{
+					Name:               "deny with token on other host",
+					Host:               "other-host.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer1,
+					ExpectResponseCode: 403,
+				},
+				{
+					Name:               "allow healthz",
+					Host:               "example.com",
+					Path:               "/healthz",
+					ExpectResponseCode: 200,
+				},
+			}
+
+			for _, c := range ingTestCases {
+				echotest.New(t, apps.All).
+					SetupForDst(func(t framework.TestContext, dst echo.Instances) error {
+						policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+							file.AsStringOrFail(t, "testdata/requestauthn/ingress.yaml.tmpl"),
+							map[string]string{
+								"Namespace": ns.Name(),
+								"dst":       dst[0].Config().Service,
```

Comment:
Is this even testing anything? this test is for the JWT on ingress gateway, it does not matter whether the backend pod (after ingress gateway) is a k8s pod or vm.

You really should add a different test for the connectivity between ingress and vm, it has nothing to do with the JWT policy here.

Reply:
in other words, does it make any sense to replace the backend pod with vm in this test?

Reply:
This test change covers the communication between ingress gateway and vm workloads, the purpose is to make sure a user flow (end user-->ingress gateway-->vm workload) is covered. Once we support ingress gateway on VM, we will update the test for the ingress gateway deployment as well.